### ISSUE TITLE
🚀  `2.6.0b1` checklist ✅ 

### ISSUE THREAD

**Issue by sydney-runkle** [pydantic/pydantic#8549]
Collection of issues to complete + PRs to finish before the upcoming minor release:

- [x] https://github.com/pydantic/pydantic/pull/8243
- [x] https://github.com/pydantic/pydantic/pull/8537
- [x] https://github.com/pydantic/pydantic/pull/8163
- [x] https://github.com/pydantic/pydantic/pull/8209 - highly desirable (added by @samuelcolvin, cc @alexmojaki)
- [x] https://github.com/pydantic/pydantic/pull/8552, which depends on https://github.com/pydantic/pydantic-core/pull/1163
- [x] https://github.com/pydantic/pydantic/pull/8567
- [x] Roll over issues from 2.6.0 milestone to 2.7.0 if they have not yet been completed

Optional:

- [x] https://github.com/pydantic/pydantic-core/pull/1162 would be nice to have (@samuelcolvin)

As a prerequisite, we'll do a V1 patch release, for which we'd like to finish these:

- [x] https://github.com/pydantic/pydantic/pull/8540
- [x] https://github.com/pydantic/pydantic/pull/8407

**Comment by hramezani** [comment]
If we want to do a V1 patch release, I suggest doing it before V2 2.6 release then we can include the latest V1 in `pydantic.v1` namespace.

**Comment by sydney-runkle** [comment]
@hramezani,

Good call. I've updated the issue description:
> As a prerequisite, we'll do a V1 patch release, for which we'd like to finish these:

**Comment by sydney-runkle** [comment]
@Viicos @romulocollopy,

I think there's some more work we need to do in order to get #6814 across the line, which is currently blocking #https://github.com/pydantic/pydantic/pull/6563.

Let's make these a priority for 2.7.0, which shouldn't come out too long after 2.6.0 🚀.

I've removed these from the list above.

**Comment by Viicos** [comment]
https://github.com/pydantic/pydantic/pull/6563 might not be blocked by https://github.com/pydantic/pydantic/pull/6814 (depending of the usage of `__name__`), but it requires some review still so if you want to have 2.6 landing soon then yes it might be worth going for 2.7!

**Comment by sydney-runkle** [comment]
@Viicos,

Let's stick with 2.7, but we focus on it soon after the release of 2.6 :). Looking forward to reviewing more thoroughly soon! 🚀

**Comment by sydney-runkle** [comment]
Everything here is done :). Marking this as closed, and will work on releasing 2.6 here shortly.

Thanks to everyone who has worked hard on this!

**Comment by einarwar** [comment]
Sorry for commenting on a closed issue. Do you have any expected release date for 2.6.0 @sydney-runkle? Some very anticipated fixes there for our project :)

**Comment by sydney-runkle** [comment]
@einarwar,

No worries at all! We're releasing it today :). Keep an eye out for a tweet soon!
